systemFlake.nix

```diff
@@ -25,25 +25,29 @@
 }@args:
 
 let
-  evalHostArgs =
-    { channelName ? "nixpkgs"
+  validateHostAttrs = {
+      channelName ? null
+    , system ? null
     , modules ? []
-    , system ? defaultSystem
     , extraArgs ? {}
-    , ...
-    }: defaultHostAttrs
-      // { 
-        inherit channelName system; 
-        modules = sharedModules ++ modules;
-        extraArgs = sharedExtraArgs // extraArgs;
-      };
+  } @ args: { inherit channelName system modules extraArgs; };
+
+  mergeHostAttrs = defaultAttrs: attrs: let
+    defaultTo = lhs: rhs: if rhs != null then rhs else lhs;
+  in
+  {
+    channelName = defaultTo (defaultTo "nixpkgs" defaultAttrs.channelName) attrs.channelName;
+    system = defaultTo (defaultTo defaultSystem defaultAttrs.system) attrs.system; # replace deaultSystem with x86_64-linux
+    modules = modules ++ defaultAttrs.modules ++ sharedModules;
+    extraArgs = extraArgs // defaultAttrs.extraArgs // sharedExtraArgs;
+  };
```

Comment:
```suggestion
evalHostArgs = 
  { channelName ? "nixpkgs"
  , system ? defaultSystem
  , modules ? [ ]
  , extraArgs ? {}
  }:  
  { 
    inherit channelName system; 
    modules = sharedModules ++ defaultHsotAttrs.modules ++ modules;
    extraArgs = sharedExtraArgs // defaultHostAttrs.extraArgs // extraArgs;
  }
```

Reply:
obsoletes `validateHostArgs` since we can validate in same function.

Reply:
I don't want external API contract enforcement be obsoleted, or made implicit, for that matter.